common/src/main/scala/common/util/IntrospectableLazy.scala

```diff
@@ -0,0 +1,32 @@
+package common.util
+
+import scala.language.implicitConversions
+
+// Adapted from https://stackoverflow.com/a/17057490/818054
+
+object IntrospectableLazy {
+
+  def lazily[A](f: => A): IntrospectableLazy[A] = new IntrospectableLazy(f)
+
+  implicit def evalLazy[A](l: IntrospectableLazy[A]): A = l()
+
+}
+
+class IntrospectableLazy[A] private(f: => A) {
```

Comment:
Maybe add some docs here on:

 - When you would anticipate this being used
 - An examples of using it
 - Why you had to do it like this (when it looks like you just way went out of your way just to reimplement `lazy val`)